browser/ui/webui/settings/brave_settings_clear_rewards_data_handler.cc

```diff
@@ -0,0 +1,233 @@
+
+#include <stddef.h>
+#include <vector>
+
+#include "base/stl_util.h"
+#include "base/values.h"
+#include "brave/browser/rewards_data/brave_rewards_data_remover_delegate.h"
+#include "brave/browser/rewards_data/counters/rewards_data_counter_factory.h"
+#include "brave/browser/rewards_data/counters/rewards_data_counter_utils.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "brave/components/brave_rewards/browser/rewards_service_factory.h"
+#include "brave/components/rewards_data/core/pref_names.h"
+#include "chrome/browser/browsing_data/browsing_data_important_sites_util.h"
+#include "chrome/browser/browsing_data/chrome_browsing_data_remover_delegate_factory.h"
+#include "components/prefs/pref_member.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browsing_data_filter_builder.h"
+#include "content/public/browser/web_ui.h"
+
+using RewardsDataType = rewards_data::RewardsDataType;
+
+namespace {
+
+const char* kCounterPrefs[] = {
+  rewards_data::prefs::kRewardsAutoContributeSites
+};
+
+} // namespace
+
+namespace settings {
+
+// ClearRewardsDataHandler ----------------------------------------------------
+
+ClearRewardsDataHandler::ClearRewardsDataHandler(content::WebUI* webui)
+    : profile_(Profile::FromWebUI(webui)),
+      weak_ptr_factory_(this) {
+  Profile* profile = Profile::FromWebUI(webui);
+  rewards_service_ =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile);
+  if (rewards_service_) {
+    rewards_service_->AddObserver(this);
+  }
+}
+
+ClearRewardsDataHandler::~ClearRewardsDataHandler() {
+  if (rewards_service_) {
+    rewards_service_->RemoveObserver(this);
+  }
+}
+
+void ClearRewardsDataHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "clearRewardsData",
+      base::BindRepeating(&ClearRewardsDataHandler::HandleClearRewardsData,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "initializeClearRewardsData",
+      base::BindRepeating(&ClearRewardsDataHandler::HandleInitialize,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "getIsContributionInProgress",
+      base::BindRepeating(
+                &ClearRewardsDataHandler::HandleGetIsContributionInProgress,
+                          base::Unretained(this)));
+}
+
+void ClearRewardsDataHandler::OnJavascriptAllowed() {
+  DCHECK(counters_.empty());
+  for (const std::string& pref : kCounterPrefs) {
+    AddCounter(RewardsDataCounterFactory::GetForProfileAndPref(profile_, pref),
+        rewards_data::ClearRewardsDataTab::REWARDS);
+  }
+}
+
+void ClearRewardsDataHandler::OnJavascriptDisallowed() {
+  weak_ptr_factory_.InvalidateWeakPtrs();
+  counters_.clear();
+}
+
+void ClearRewardsDataHandler::HandleClearRewardsDataForTest() {
+  // HandleClearRewardsData takes in a ListValue as its only parameter. The
```

Comment:
so why aren't you creating the ListValue in the test? The `ForTest` methods are typically to do something that otherwise can't be done through the existing methods

Reply:
This method was a copy-over and has been removed.